### PR TITLE
Query NFTs by collection

### PR DESCRIPTION
**Pull request by imabdulbasit**
- Collection root query which takes collection nft address as parameter
- collection data loader added in Nft

### CODE REVIEW

**Review by ray-kast — APPROVED** [review]

File: crates/graphql/src/schema/objects/collection.rs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    pub fn image(&self) -> Option<&str> {
+        self.image.as_deref()
+    }
+
+    pub async fn nfts(&self, ctx: &AppContext) -> FieldResult<Vec<CollectionNft>> {
+        ctx.collection_nfts_loader
+            .load(self.address.clone().into())
+            .await
+            .map_err(Into::into)
+    }
```
Comment (kespinola) [inline crates/graphql/src/schema/objects/collection.rs:64]: Need to get limit and offset as arguments for this field or going to be returning thousands of nfts.

**Review by kespinola — CHANGES_REQUESTED** [review]
Like to release with limits and offset for the nfts field on collections.

**Comment by kespinola** [comment]
Chatted in standup Thursday but for v1 of filtering nfts by collections about allowing subquery but returning all the addresses of nfts belonging to a collection to filter the existing query is going result of 10K plus address being returned for a pfp project for example. Does this sound right @imabdulbasit?

**Comment by imabdulbasit** [comment]
> Chatted in standup Thursday but for v1 of filtering nfts by collections about allowing subquery but returning all the addresses of nfts belonging to a collection to filter the existing query is going result of 10K plus address being returned for a pfp project for example. Does this sound right @imabdulbasit?

We already pass collection mint address as filter. So if the filter passes collection metadata address, we need the mint address and for that we need a separate query which will return one mint address of a collection and use the same filter like we have.

**Review by ray-kast — CHANGES_REQUESTED** [review]
Branch needs to be rebased to accommodate a new CI step.

**Pull request closed without merging**